row.py

```diff
@@ -988,9 +980,131 @@ def join_spreadsheet_info(filtered_results_file, out_dir):
+    working_df["text"] = working_df.apply(lambda r: r["text"].replace("\\", "").strip(), axis=1)
+    working_df["text"] = working_df.apply(lambda r: r["text"].replace("::", ":").strip(), axis=1)
+
+    #: remove leading colons
+    logging.info("Removing leading colons")
+    working_df["text"] = working_df["text"].apply(lambda x: x.lstrip(":"))
+
+    #: replace euro symbol with 'E'
+    logging.info("Replacing euro symbol with 'E'")
+    working_df["text"] = working_df.apply(lambda r: r["text"].replace("€", "E").strip(), axis=1)
+
+    #: replaced accented characters with non-accented characters
+    logging.info("Replacing accented characters with non-accented characters")
+    working_df["text"] = working_df.apply(
+        lambda r: "".join(c for c in unicodedata.normalize("NFD", r["text"]) if unicodedata.category(c) != "Mn"), axis=1
+    )
+
+    #: flag parcels that contains special characters
+    working_df["special_char"] = "pass"
+    mask = working_df["text"].str.contains(r"[^\w:]") == True
+    working_df.loc[mask, "keep"] = "no"
+    working_df.loc[mask, "special_char"] = "fail"
+    logging.info("Number parcels with special characters flagged: %i", mask.value_counts()[1])
+
+    #: flag parcels that start with a letter or non-digit, should start with a number
+    working_df["nondigit_start"] = "pass"
+    mask = ~working_df["text"].str.contains(r"^\d") == True
+    working_df.loc[mask, "keep"] = "no"
+    working_df.loc[mask, "nondigit_start"] = "fail"
+    logging.info("Number parcels starting with a non-digit flagged: %i", mask.value_counts()[1])
+
+    #: flag parcels with a ':P' pattern; permits, not relevant - but PUE is okay!
+    working_df["permit"] = "pass"
+    mask = working_df["text"].str.contains(r"(?!:PUE):P") == True
+    working_df.loc[mask, "keep"] = "no"
+    working_df.loc[mask, "permit"] = "fail"
+    logging.info("Number of permits (:P pattern, but not :PUE) flagged: %i", mask.value_counts()[1])
+
+    #: flag parcels with a colon, that do not have a number before the colon
+    working_df["no_number_before_colon"] = "pass"
+    mask = working_df["text"].str.contains(r"^[^0-9]*:") == True
+    working_df.loc[mask, "keep"] = "no"
+    working_df.loc[mask, "no_number_before_colon"] = "fail"
+    logging.info("Number of colons not preceeded by a number flagged: %i", mask.value_counts()[1])
+
+    #: flag parcels longer than 13 characters
+    working_df["too_long"] = "pass"
+    mask = working_df["text"].str.len() > 13
+    working_df.loc[mask, "keep"] = "no"
+    working_df.loc[mask, "too_long"] = "fail"
+    logging.info("Number of parcels exceeding 13 characters flagged: %i", mask.value_counts()[1])
+
+    #: flag parcels with 4 or more letters given that no colon exists (or 1 or 8)
+    working_df["too_many_letters"] = "pass"
+    mask = (working_df["text"].str.contains("^[^:18]*$")) & (working_df["text"].str.contains("(?:[A-Z][^A-Z]*){4,}"))
```

Comment:
**question**: what is going on here? will you add a couple cases in comments in the form of

```py
#: value = pass
#: value = fail
```

for all these regexes please

Reply:
do you even regex, bro?

Reply:
I'm adding some examples.

Reply:
It looks like it flags all of them to me

![image](https://user-images.githubusercontent.com/325813/228616552-740904fc-4e8f-41d8-8e58-855f460117c9.png)

Reply:
Oops, when I was checking the `^[^:18]*$|(?:[A-Z][^A-Z]*){4,}` regex it had space on the end when I pasted and that screwed it up.